apps/web/src/components/providers/AuthProvider.tsx

```diff
@@ -25,16 +25,16 @@ export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
     return <EnterpriseAuthProvider>{children}</EnterpriseAuthProvider>;
   }
 
-  // eslint-disable-next-line react-hooks/rules-of-hooks
   return <CommunityAuthProvider>{children}</CommunityAuthProvider>;
 };
 
 export const useAuth = () => {
-  const context = IS_EE_AUTH_ENABLED ? EnterpriseAuthContext : CommunityAuthContext;
-  const value = useContext(context);
+  const ctx = IS_EE_AUTH_ENABLED ? EnterpriseAuthCtx : CommunityAuthCtx;
+  const context = useContext(ctx);
```

Comment:
Can't use `useEnterpriseAuthCtx` from the `createContextAndHook` utility here because that would result in usage like this:

```typescript
const value = IS_EE_AUTH_ENABLED ? useEnterpriseAuthCtx() : useCommunityAuthCtx()
```

Which triggers both hooks and violates ESLint rule of hooks